src/Resources/Timeline.php

```diff
@@ -0,0 +1,262 @@
+<?php
+
+namespace SevenShores\Hubspot\Resources;
+
+use DateTime;
+
+class Timeline extends Resource
+{
+    /**
+     * Create or Update Timeline Event
+     *
+     * @param int           $appId
+     * @param int           $eventTypeId
+     * @param string        $id
+     * @param int|null      $objectId
+     * @param string|null   $email
+     * @param string|null   $utk
+     * @param array         $extraData
+     * @param DateTime|null $timestamp
+     *
+     * @return mixed
+     *
+     * @see http://developers.hubspot.com/docs/methods/timeline/create-or-update-event
+     */
+    public function createOrUpdate(
+        $appId,
+        $eventTypeId,
+        $id,
+        $objectId = null,
+        $email = null,
+        $utk = null,
+        $extraData = [],
+        DateTime $timestamp = null
+    ) {
+        $endpoint = "https://api.hubapi.com/integrations/v1/{$appId}/timeline/event";
+
+        $data['json'] = [
+            'eventTypeId' => $eventTypeId,
+            'id'          => $id,
+            'objectId'    => $objectId,
+            'email'       => $email,
+            'utk'         => $utk,
+            'extraData'   => $extraData,
+        ];
+
+        if (isset($timestamp)) {
+            // Times must be in milliseconds epoch time.
+            $data['json']['timestamp'] = $timestamp->getTimestamp() * 1000;
+        }
```

Comment:
Yeah, good point.

I kept sending seconds before I realized in the docs it said to use milliseconds instead of seconds, and that's what made me decide to go with a `DateTime` so other people wouldn't make that mistake.

Your solution is nice for any dates before `Sat, 20 Nov 2286 17:46:39 GMT` (`9999999999`), but after that date we'll have a problem. (Not sure if anyone will be using this PHP library in the year 2286...haha)

Maybe something more like this:

```php
if ($timestamp instanceof \DateTime) {
    $timestamp = $timestamp->getTimestamp() * 1000;
}

if (isset($timestamp)) {
    $data['json']['timestamp'] = $timestamp;
}
```

That way the function can support both a `DateTime` and maybe in the PHPDoc we just mention that it either has to be a `DateTime` or an already converted to milliseconds timestamp. Could even rename it to be more like this:

```php
/**
 * @param mixed $timestampMilliseconds - Timestamp in milliseconds format or DateTime
 */
```

Thoughts?